6.bulk_Morphology_Elastic_Network/3.model_coefficients/scripts/1.model_coefficients.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pathlib
+
+import joblib
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+
+# In[2]:
+
+
+cell_type = "SHSY5Y"
+
+
+# In[3]:
+
+
+# set import paths
+model_path = pathlib.Path(
+    f"../../1.train_models/models/regression/{cell_type}/aggregated_with_nomic/IL-1 beta [NSU]_final__all_nomic.joblib"
+)
+
+# set and create output paths
+output_path = pathlib.Path(f"../results/regression/{cell_type}/")
+output_path.mkdir(parents=True, exist_ok=True)
+
+
+# In[4]:
+
+
+# set path of models
+model_path = pathlib.Path(
+    f"../../1.train_models/models/regression/{cell_type}/aggregated_with_nomic/"
+)
+# get the list of files in the directory
+model_files = list(model_path.glob("*.joblib"))
+for model_file in model_files:
+    pass
+model_file
+
+
+# In[5]:
+
+
+for model_file in model_files:
+    # import model
+    model = joblib.load(model_file)
+    # create a df with the coefficients
+    df_coefficients = pd.DataFrame(
+        model.coef_, index=model.feature_names_in_, columns=["coefficients"]
+    )
+    # sort by absolute value of coefficients
+    df_coefficients = df_coefficients.reindex(
+        df_coefficients["coefficients"].abs().sort_values(ascending=False).index
+    )
+
+    df_coefficients.head(10)
```

Comment:
Double checking: does this call need a print for appearance in the output of the cell (would it disappear in the processing of results)?

Reply:
It would need a print statement, but I ended up removing it!